skrub/_table_vectorizer.py

```diff
@@ -6,27 +6,31 @@
 
 import warnings
 from collections import Counter
-from typing import Literal
 
 import numpy as np
 import pandas as pd
 import sklearn
-from numpy.typing import ArrayLike
 from pandas._libs.tslibs.parsing import guess_datetime_format
 from pandas.core.dtypes.base import ExtensionDtype
 from scipy import sparse
-from sklearn.base import TransformerMixin, clone
+from sklearn.base import BaseEstimator, TransformerMixin, clone
 from sklearn.compose import ColumnTransformer
 from sklearn.compose._column_transformer import _get_transformer_list
 from sklearn.preprocessing import OneHotEncoder
 from sklearn.utils import Bunch
-from sklearn.utils.deprecation import deprecated
-from sklearn.utils.metaestimators import _BaseComposition
 from sklearn.utils.validation import check_is_fitted
 
 from skrub import DatetimeEncoder, GapEncoder
 from skrub._utils import parse_astype_error_message
 
+HIGH_CARDINALITY_TRANSFORMER = GapEncoder(n_components=30)
+LOW_CARDINALITY_TRANSFORMER = OneHotEncoder(
+    sparse_output=False,
```

Comment:
we expose the `ColumnTransformer`'s `sparse_threshold` parameter but with our default transformers the default will always be dense (even if toe onehot encoder yields many zeros)

we could consider
- pointing out in the doc that users need to change the transformers if they want sparse output
- not exposing the `sparse_threshold` and always returning dense data
- making the onehot encoder sparse by default

(not in this PR)